### PR TITLE
[RFR] Kebab  Widget

### PR DESCRIPTION
**Pull request by digitronik**
- Moving kebab widget from [miq](https://github.com/ManageIQ/integration_tests/blob/master/cfme/dashboard.py#L23)
- Changes:
    - Added `locator` as on of optional agrument. So if id not match and want to select `kebab` other than first. 
    - As its `dropdown` type  of  widget so methods should  similar to dropdown. renamed `select` method with `item_select`.
    - Added `has_item` method

- Added test

### CODE REVIEW

**Comment by digitronik** [comment]
related to https://github.com/RedHatQE/widgetastic.patternfly/issues/75

**Comment by digitronik** [comment]
Indexing is just trial/proposed idea; else I will revert this...

**Comment by digitronik** [comment]
@mshriver: After understanding the concepts of `inheritance` as per your suggestions on RC l.e. In the future if someone like to inherit `Kebab`and try to change locators as per project requirement. 

We are providing the main two options:
1. `id`:
       If a widget has a specific id then we can go with this option. For extension means, if anyone inherited and want to change then a user can just change `BASE_LOCATOR`

2. `locator`:
      `locator` is an important argument as per my observation. Let suppose, we won't have `Id` then we can go with the button locator. I would like every new widget has a locator option.

3. `index`: it's completely `optional`. But, I like to introduce as per current implementation. It not going to effect on main widget usage. It's pointing to `patternfly kebab` `standard implementation`.

trying to explain indexing: `(which is optional)` (With Milan local implementation)
- Suppose; I have a page having only one kebab like `SSUI`:
`x = Kebab(view)`   `# It will give you first kebab on a page. It just like PaginationPlane view`
`Note`: the other two argument options available. the user can use them.

- Suppose; I have three kebabs on the same page: Like `XPath` we can check for `index` while widget initialization. 
`x=Kebab(view, index=3)`

I understand `index` is not the proper way, but we can not say it not a `practical` approach. As `id` or `locator` changes as per development changes. It should be one option. Users can decide with which they like to go.

File: src/widgetastic_patternfly/__init__.py [hunk 1]
```diff
@@ -2249,3 +2249,82 @@ class AggregateStatusMiniCard(AggregateStatusCard):
...
+    """
+
+    ROOT = ParametrizedLocator("{@locator}")
+    BASE_LOCATOR = ".//div[contains(@class, 'dropdown-kebab-pf') and ./button[@id={}]]"
+    UL = './ul[contains(@class, "dropdown-menu")]'
+    BUTTON = "./button"
+    ITEM = "./ul/li/a[normalize-space(.)={}]"
+    ITEMS = "./ul/li/a"
+
+    def __init__(self, parent, id=None, locator=None, index=1, logger=None):
```
Comment (mshriver) [inline src/widgetastic_patternfly/__init__.py]: I don't think this behavior should exist in this widget definition. providing a broad 'locator' parameter covers this use case, but I don't want to encourage that use.

Comment (mshriver) [reply src/widgetastic_patternfly/__init__.py]: The DOM could change, with kebabs removed or inserted, breaking the expectations of the locator defined by index.  There would be no way to tell you're interacting with a different widget than the last one.

**Comment by digitronik** [comment]
@mshriver removed indexing completely... :+1:

**Pull request merged**